src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -94,6 +94,12 @@ public abstract partial class WebRequestPSCmdlet : PSCmdlet
         [Parameter]
         public virtual SwitchParameter SkipCertificateCheck { get; set; }
 
+        /// <summary>
+        /// gets or sets the CertificateValidationScript property
+        /// </summary>
```

Comment:
In that case, perhaps the two parameters should be in mutually exclusive parameter sets?

Reply:
@rkeithhill  We don't want these to be full mutually exclusive via parameter set. we want to support thuis usage:

```powershell
$script = { <# some validation script #> }
$PSDefaultParameterValues['Invoke-RestMethod:CertificateValidationScript'] = $script

# All calls will use the validation script
Invoke-RestMethod https://unknonwnurl.com/

# But the user may want to override their normal validation for a certain Url
Invoke-RestMethod https://devsite.com/ -SkipCertificateCheck
```

Also, the Web Cmdlets have traditionally had a single parameter set and there are other parameters that are exclusive (i.e. `-Infile` and `-Body`). Introducing a parameter set here would be awkward.

@iSazonov I can update the summary, but other parameters with this kind of behavior to not call it out in their summaries. I believe these summaries are only visible when coding in C#, correct? Since these are `PSCmdlet` derived, you don't even see these summaries from with in the same assembly. In which case, this is visible only when working in classes that derive from `WebRequestPSCmdlet`.

Reply:
If Web Cmdlets have traditionally had a single parameter set I agree to follow the pattern.

My thought was that we should add the comment somewhere. The place is best.  It's good for documenting code. Also we generate XML help files - I guess it can be used for creation of HTLM documentation.

Reply:
If I'm not mistaken, the help files are generated from the [PowerShell/PowerShell-Docs](https://github.com/PowerShell/PowerShell-Docs) markdown? I can certainly say that I have never seen `gets or sets the Body property` (the summary for `-Body`) in anything. I certainly don't see any of the parameter summaries in any of the help xmls. The only place this shows up is the `Microsoft.PowerShell.Commands.Utility.xml` file and that has the same unhelpful summaries as the other parameters.

I will update it. I just think it wont be actually that helpful to anyone.

Reply:
There's two types of help, there's the updatable help that exists in PowerShell-Docs and there's api reference docs that get generated from the assemblies.  Any public api should be documented in code as the documentation is generated from that xml.

Reply:
@SteveL-MSFT Is this the documentation you are referring to? https://msdn.microsoft.com/en-us/library/microsoft.powershell.commands.webrequestpscmdlet(v=vs.85).aspx If not, what exactly? I ask because the other summaries are very unhelpful and if these are being documented somewhere they could use some serious attention. I have never seen these summaries outside of the XML file and the source code.

Reply:
We already add in our coding guigance that we welcome good comments everywhere. I hope in time we'll see a well-documented code 😄